### PR TITLE
OPRUN-3021: Improve README

### PR DESCRIPTION
**Pull request by tmshort**
Add additional information to the README files.
Add help for the sync scripts

### CODE REVIEW

**Review by stevekuznetsov — APPROVED** [review]
Looks awesome, thank you!

**Review by dinhxuanvu — APPROVED** [review]
/approve

File: scripts/README.md [hunk 1]
```diff
@@ -2,4 +2,71 @@
...
+Then run the following:
+```sh
+scripts/sync_pop_candidate.sh -a sync
+```
+The commits in the `sync.cherrypick` file will be applied in the specified
+order.
+
+Even if you only have a single commit, this procedure will follow the same
+process that `sync.sh` does, to ensure no steps are missed.
+
```
Comment (grokspawn) [inline scripts/README.md:67]: Doesn't sync_pop_candidate.sh also make manifests, vendor, etc. in ways that sync.sh does not?  
(And also, sync_pop_candidate.sh needs to make verify-manifests in order to satisfy CI.)

**Review by grokspawn — CHANGES_REQUESTED** [review]

**Comment by dinhxuanvu** [comment]
@tmshort I have no problem with merging it manually as it is just a doc change.

File: scripts/README.md (hunk 1 above)
Comment (tmshort) [reply scripts/README.md:67]: I'm going to document the need to do a `make verify` but fixing the downstreaming scripts to do that is a bit more involved, so I'd rather save that for another PR.

Comment (grokspawn) [reply scripts/README.md:76]: My point was really more about the statement about 'will follow the same process' when there is a behavior gap between the scripts. 
for e.g.:
- [make manifests](https://github.com/openshift/operator-framework-olm/blob/9a977fa7b6c40ad237abc4645dc41e58cafbb944/scripts/sync_pop_candidate.sh#L109)
- [updating commit format to track commit sources](https://github.com/openshift/operator-framework-olm/blob/9a977fa7b6c40ad237abc4645dc41e58cafbb944/scripts/sync_pop_candidate.sh#L108)

Comment (tmshort) [reply scripts/README.md:76]: But _it is_ the same process. The sync.sh script:

1. The upstream repositories are added as remotes to the monorepo.
2. The upstream repositories are fetched.
3. A merge branch with the current date (sync-YYYY-MM-DD) is created to do the bulk sync.
4. Create a list of candidate commits from upstream to downstream via `sync_get_candidates.sh`
5. Apply each commit via `sync_pop_candidate.sh`

The `sync.sh` script literally calls `sync_pop_candidate.sh` to do all the work for a list of commits (generated by `sync_get_candidates.sh`.

The problem is that there's additional verification steps via `make verify`/`make verify-manifests`, that need to occur.

Comment (grokspawn) [reply scripts/README.md:76]: Argh!  You're completely correct.  I missed the call to sync_pop_candidates.sh. ☹️ 
Thanks for the sanity.

Comment (grokspawn) [reply scripts/README.md:76]: IMHO we should solve the verification "additional steps" problem in changes to sync_pop_candidates.sh in another PR.

**Review by grokspawn — APPROVED** [review]

File: scripts/README.md (hunk 1 above)
Comment (tmshort) [reply scripts/README.md:78]: Agreed.

**Comment by stevekuznetsov** [comment]
FWIW Tide will re-test even when there are overrides but yea seems fine to override

**Comment by stevekuznetsov** [comment]
Also can configure e2e to skip docs

**Comment by grokspawn** [comment]
/lgtm

re-lgtm-ing

**Comment by tmshort** [comment]
(looks as though I can't override)

**Comment by tmshort** [comment]
/retest
Or could an OWNER `/override e2e-gcp-olm` ?

**Pull request merged**